### DISCUSSION TITLE
Cross-platform by design?

### DISCUSSION THREAD

**Discussion by rexendevar** [free-audio/clap#226] [Q&A] [answered]
Just wondering if CLAP is intended to be cross-platform by design. As a Linux user it would be extremely nice to have a plugin designed on Windows or Mac work without having to use WINE or something. If this is not part of the design consider this a feature request.

**Comment by baconpaul** [comment, marked as answer]
So it is cross platform by design in that the headers themselves can build an exportable dynamic library on a wide variety of operating systems, including linux.

Whether a plugin manufacturer choses to avail themselves of that opportunity is another question. Today plugins are written in C++ which is system dependent, requires build processes, and so on. While there are some emerging technologies which could perhaps allow you to truly be write-once-run-multiple at realtime rate, they are not how audio software works today, although I know folks in CLAP team are looking at those emerging options.

Plenty of plugins designed and developed on mac work fine on linux. Surge is such a plugin! But we do have to do extra compile steps. Not every vendor choses to do that, and in some of the platforms people use on top of clap, not every platform supports it.

But if you want to run a plugin in linux bitwig right now the choices are (1) ask the plugin vendor to port it to linux; CLAP allows them to do that easily but other factors in their stack might not or (2) run yabridge

**Reply by rexendevar** [reply]
Is it any easier to create a Linux version using CLAP than other standards (e.g. VST3)?

**Reply by baconpaul** [reply]
The answer to that depends on many things. Vst3 works fine on linux so making a Linux clap is easier than making a Linux vst3 only inasmuch as making a clap is easier than making a vst3. Most folks who code directly to the api have found the clap api easy to use

but the plugin layer is a very small part of your average plugin. Clap is lovely i thst it is compact light portable and mostly gets out of your way. But if you have a plugin which is a vst3 that uses directx to render, turning into a clap that uses directx still means you have to rewrite the graphics to port to linux

so my answer to your question is “yes definitely, but that’s not the critical blocker for Linux adoption for many plugins most of the time”.

**Reply by rexendevar** [reply]
Thank you for giving such a detailed response.
